src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -799,13 +799,46 @@ AdjustDistributedExecutionAfterLocalExecution(DistributedExecution *execution)
 
 /*
  * ExecuteUtilityTaskListWithoutResults is a wrapper around executing task
- * list for utility commands. It simply calls in adaptive executor's task
- * execution function.
+ * list for utility commands. For remote tasks, it simply calls in adaptive
+ * executor's task execution function. For local tasks (if any), kicks Process
+ * Utility via CitusProcessUtility for utility commands. As some local utility
+ * commands can trigger udf calls, this function also processes those udf calls
+ * locally.
  */
 void
-ExecuteUtilityTaskListWithoutResults(List *taskList)
+ExecuteUtilityTaskListWithoutResults(List *taskList, bool localExecutionSupported)
 {
-	ExecuteTaskList(ROW_MODIFY_NONE, taskList, MaxAdaptiveExecutorPoolSize);
+	RowModifyLevel rowModifyLevel = ROW_MODIFY_NONE;
+
+	List *localTaskList = NIL;
+	List *remoteTaskList = NIL;
+
+	bool readOnlyPlan = !TaskListModifiesDatabase(rowModifyLevel, taskList);
```

Comment:
Could we ever have `bool readOnlyPlan = false` with utility commands? 

If not, it might be better to do `bool readOnlyPlan = false`, as it is more clear.

(I noted this because some utility tasks are converted to `SELECT UDF()`, though I expect TaskListModifiesDatabase return false, still might be good to double check that case)

Reply:
I guess `TaskListModifiesDatabase` can return different results for SELECT commands & regular utility commands due to the call made to `ReadOnlyTask`

Reply:
But all the SELECT commands going through here should be considered as modification, and hence the `ReadOnlyTask ` returns that.

However, what I meant was, we can skip calling `TaskListModifiesDatabase ` to save some CPUs and simply set it `readOnlyPlan  = false`, which would be a slight improvement in readability (by adding a single line comment saying that utility commands are always modifications)

Reply:
Now I get the point, makes sense to me